week-4-practise/level-1/exercise-1.js

```diff
@@ -36,9 +36,42 @@ var writers = [
     occupation: "writer",
     age: 64,
     alive: true
-  },
+  }
 ];
 
+writers.forEach(function(writer) {
+  console.log(
+    "Hi, my name is " +
+      writer.firstName +
+      " " +
+      writer.lastName +
+      "." +
+      " I am " +
+      writer.age +
+      " years old, and work as a " +
+      writer.occupation
+  );
+});
+
 /*
 If you want an extra challenge, only `console.log()` the writers that are alive.
-*/
\ No newline at end of file
+*/
+
+var aliveWriters = writers
+  .filter(function(writer) {

```

Comment:
The use of filter is good. However you should also think of the performance consequence of using `filter` and `forEach one after the other. If you do `filter`, the program will iterate through the whole array and filter it. Then the call of `forEach` will iterate through the whole new array and `console.log`. If all writers are alive then you would need to iterate through the `writers` array 2 times. 

To avoid unnecessary iterations you could have used only the for each and have an if statement in it:

```
writers.forEach(function(writer) {
    if (writer.alive == true) {
        console.log(....)
    }
})
```

This would cause your code to iterate through the array only once. 

Not saying what you did is bad but it's good to think about performance